the-collab-lab/tcl-13-smart-shopping-list
# At rb react router

## Description
Updated dependencies, added `react-router-dom`. 
Added a Components folder for organization. 
Added Nav, AddItems, ViewList, and Router components to the Components folder.

**Persistent Links**
To ensure the "Add to List" and "View List" links were present and persistent at the bottom of the page, we added them to App as a Nav component which is currently rendered below the header. 

**URL Updates**
We nested App within the Router component, then pointed Index.js to Router instead of App, to track and update changes to the URL. `Switch` and nested `Route` tags were used to update the URL to match the component being displayed (AddItems or ViewList). 

**Corresponding Link is Bold**
In Nav, instead of `Link` we used `NavLink` so that we could use `activeStyle` to make the link that corresponds to the presently displayed component bold. 

## Related Issue
Fixes #2 

## Acceptance Criteria



## Type of Changes
|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates
- Updated dependencies, added `react-router-dom`
- Added a Components folder for organization. 
- Added Nav, AddItems, ViewList, and Router components to the Components folder.
- Index.js points to Router component instead of app
- Added `Switch` to App component, removed most of the default header styling

### Before
Default React Screen
![Screen Shot 2020-10-05 at 8 25 36 PM](https://user-images.githubusercontent.com/49924080/95145365-043c4d80-0749-11eb-8438-8ec76424ec46.png)

### After
![Screen Shot 2020-10-05 at 8 32 06 PM](https://user-images.githubusercontent.com/49924080/95145712-ffc46480-0749-11eb-89e6-8c6865020b64.png)

Using View List as an example.
![Screen Shot 2020-10-05 at 8 32 25 PM](https://user-images.githubusercontent.com/49924080/95145721-04891880-074a-11eb-96aa-c58af6b0341f.png)

URL Update
![Screen Shot 2020-10-05 at 8 32 37 PM](https://user-images.githubusercontent.com/49924080/95145733-0b179000-074a-11eb-874b-fae6ef627456.png)


## Testing Steps / QA Criteria
Please click the links "Add Items" and "View List". The URL should update and link should be bold for the corresponding component.
